src/cpp/r/RExec.cpp

```diff
@@ -132,7 +134,13 @@ Error evaluateExpressionsUnsafe(SEXP expr,
 {
    // detect if an error occurred (only relevant for EvalTry)
    int errorOccurred = 0;
-   
+
+   if (!isMainThread())
+   {
+      LOG_ERROR_MESSAGE("evaluateExpression called from thread other than main");
+      return rCodeExecutionError("Attempt to eval R on thread other than main", ERROR_LOCATION);
```

Comment:
Let's include `str` as an error property to make it clearer what we're trying to do